assets/js/cardknox.js

```diff
@@ -17,195 +17,250 @@
-                document.getElementById("ifieldsError").textContent
-              );
+      wc_cardknox_form.unblock();
+    },
+
+    onSubmit: function (e) {
+      //debugger;
+      console.log("onSubmit");
+      // wc_cardknox_form.form.validate_field();
+      if (wc_cardknox_form.isCardknoxChosen() && !wc_cardknox_form.hasExp()) {
+        e.preventDefault();
+        wc_cardknox_form.block();
+        getTokens(
+          function () {
+            //onSuccess
+            //perform your own validation here...
+            if (document.getElementsByName("xCardNum")[0].value === "") {
+              $(document).trigger("cardknoxError", "Card Number Required");
+              return false;
+            }
+            if (document.getElementsByName("xCVV")[0].value === "") {
+              $(document).trigger("cardknoxError", "CVV Required");
               return false;
-            },
-            //30 second timeout
-            30000
-          );
-          return false;
-        }
-        return true;
-      },
-  
-      onCardknoxResponse: function () {
-        var xExp = document
-          .getElementById("cardknox-card-expiry")
-          .value.replace(/\s|\//g, "");
-        if (xExp.length != 4) {
-          $(document).trigger("cardknoxError", "Invalid expiration date");
-          return false;
-        }
-        console.log("onCardknoxResponse");
-        wc_cardknox_form.form.append(
-          "<input type='hidden' class='xExp' name='xExp' value='" + xExp + "'/>"
+            }
+            console.log("Success");
+            wc_cardknox_form.onCardknoxResponse();
+          },
+          function () {
+            //onError
+            console.log("error");
+            $(document).trigger(
+              "cardknoxError",
+              document.getElementById("ifieldsError").textContent
+            );
+            return false;
+          },
+          //30 second timeout
+          30000
+        );
+        return false;
+      }
+      return true;
+    },
+
+    onCardknoxResponse: function () {
+      var xExp = document
+        .getElementById("cardknox-card-expiry")
+        .value.replace(/\s|\//g, "");
+      if (xExp.length != 4) {
+        $(document).trigger("cardknoxError", "Invalid expiration date");
+        return false;
+      }
+
+      // Extract the month and year from the expiration date
+      let month = parseInt(xExp.substr(0, 2));
+      let year = parseInt(xExp.substr(2));
+
+      // Validate the expiration month and year
+      let currentDate = new Date();
+      let currentYear = currentDate.getFullYear() % 100; // Get the last two digits of the current year
+      let currentMonth = currentDate.getMonth() + 1; // January is month 0 in JavaScript
+
+      if (
+        year < currentYear ||
+        (year === currentYear && month < currentMonth)
+      ) {
+        $(document).trigger(
+          "cardknoxError",
+          "Incorrect credit card expiration date."
```

Comment:
I feel like we keep going back and forth on this. Do you want to discuss?
```suggestion
          "Expiration must be in the future"
```